react/src/hooks/useMemoWithPrevious.tsx

```diff
@@ -0,0 +1,36 @@
+import { DependencyList, useEffect, useMemo, useRef, useState } from 'react';
+
+export const useMemoWithPrevious = <T,>(
+  factory: () => T,
+  { deps, initialPrev }: { deps: DependencyList; initialPrev?: T },
+) => {
+  const prevRef = useRef(initialPrev);
+  const [prevResetKey, setPrevResetKey] = useState({});
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
```suggestion
```

Reply:
Without this eslint complained `React Hook useMemo has a missing dependency: 'factory'. Either include it or remove the dependency array.eslintreact-hooks/exhaustive-deps